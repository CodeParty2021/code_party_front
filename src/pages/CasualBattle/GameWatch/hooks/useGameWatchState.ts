import { useEffect } from "react";
import { useNavigate } from "react-router-dom";

import { useRoomSync } from "hooks/RoomSyncHooks/useRoomSync";
import {
  ResultType,
  useFetchResult,
} from "hooks/ResultAPIHooks/useFetchResult";
import { useFetchJson } from "hooks/ResultAPIHooks/useFetchJson";

export type IResponse = {
  isAnalyzing: boolean;
  analyzingError: boolean;
  result?: ResultType;
  json?: string;
  exitBtnHandler: () => void;
};

export const useGameWatchState = (): IResponse => {
  const { room, updateMember } = useRoomSync();
  const { data: result, fetchResult } = useFetchResult();
  const { data: json, fetchJson } = useFetchJson();
  const navigate = useNavigate();

  //初期処理
  useEffect(() => {
    //ユーザ状態を更新
    updateMember({
      status: "watching",
    });
  }, []);

  // 何らかの理由でルームから出された場合はロビーに戻る
  useEffect(() => {
    if (!room.isEntered) {
      navigate("/casual-battle");
    }
  }, [room.isEntered]);

  //シミュレーション結果が更新されたら結果をフェッチする
  useEffect(() => {
    if (room.info?.analyzingResult?.resultId) {
      fetchResult(room.info?.analyzingResult.resultId);
      fetchJson(room.info?.analyzingResult.resultId);
    }
  }, [room.info?.analyzingResult]);

  const _exitBtnHandler = () => {
    navigate("/casual-battle/waiting-room");
  };

  return {
    isAnalyzing: !room.info?.analyzingResult?.resultId,
    analyzingError: Boolean(room.info?.analyzingResult?.error),
    result: result,
    json: json,
    exitBtnHandler: _exitBtnHandler,
  };
};
