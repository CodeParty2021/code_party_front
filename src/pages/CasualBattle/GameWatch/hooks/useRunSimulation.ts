import { useEffect } from "react";

import { useRoomSync } from "hooks/RoomSyncHooks/useRoomSync";
import { useRunCodes } from "hooks/CodeAPIHooks/useRunCodes";

export type IResponse = {};

export const useRunSimulation = (): IResponse => {
  const { room, isHost, updateRoomInfo } = useRoomSync();
  const {
    data: simulationResult,
    error: simulationError,
    runCodes,
  } = useRunCodes();

  //初期処理
  useEffect(() => {
    //シミュレーションの実行(ホストのみ実行)
    if (isHost) {
      const codes: string[] = [];
      room.sortedKeysOfMembers.forEach((key) => {
        const member = room.members[key];
        if (member.codeId) codes.push(member.codeId);
      });
      runCodes(...codes);
    }
  }, []);

  //シミュレーション結果を受信した場合
  useEffect(() => {
    if (simulationResult) {
      updateRoomInfo({
        analyzingResult: {
          resultId: simulationResult.jsonId,
        },
      });
    }
  }, [simulationResult]);

  //シミュレーション中にエラーが発生した場合
  useEffect(() => {
    if (simulationError) {
      updateRoomInfo({
        analyzingResult: {
          error: "シミュレーション実行中にエラーが発生しました。",
        },
      });
    }
  }, [simulationError]);

  return {};
};
