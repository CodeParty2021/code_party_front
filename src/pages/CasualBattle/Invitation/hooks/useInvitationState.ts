import { useEffect, useState } from "react";
import { Params, useNavigate, useParams } from "react-router-dom";
import { useRoomSync } from "hooks/RoomSyncHooks/useRoomSync";
import { RootState } from "store";
import { useSelector } from "react-redux";

export type IResponse = {
  isLogin: boolean;
  errorMessage?: string;
  roomId?: string;
  enterBtnDisabled: boolean;
  enterBtnClickHandler: () => void;
};

export const useInvitationState = (): IResponse => {
  const { roomId } = useParams<Params<string>>();
  const { isLogin } = useSelector((state: RootState) => state.user);
  const [enterBtnDisabled, setEnterBtnDisabled] = useState(false);
  const { room, enterRoom } = useRoomSync();
  const [errorMessage, setErrorMessage] = useState<string | undefined>(
    undefined
  );
  const navigate = useNavigate();

  useEffect(() => {
    if (room.isEntered) {
      navigate("/casual-battle/waiting-room");
    }
  }, [room.isEntered]);

  //入場ボタン
  const _enterBtnHandler = () => {
    if (typeof roomId === "string") {
      setEnterBtnDisabled(true);
      enterRoom(roomId).catch(() => {
        setEnterBtnDisabled(false);
      });
    } else {
      setErrorMessage("招待リンクが無効です。");
    }
  };

  return {
    isLogin: isLogin,
    errorMessage: errorMessage,
    roomId: roomId,
    enterBtnDisabled: enterBtnDisabled,
    enterBtnClickHandler: _enterBtnHandler,
  };
};
