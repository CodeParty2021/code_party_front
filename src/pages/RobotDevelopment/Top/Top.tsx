import React from "react";
import { Link } from "react-router-dom";

type Prop = {};

export const RobotDevelopmentTop: React.FC<Prop> = () => {
  return (
    <div>
      <h1>ロボット開発モードトップ画面</h1>
      <Link to="/tutorial/world/:id/missions">
        チュートリアルモードへ進む(world選択画面は画面遷移図に基づき一旦飛ばす)
      </Link>
      <Link to="/garage">ガレージモード</Link>
      <Link to="/robot-development/specification">プログラム仕様書</Link>
    </div>
  );
};
