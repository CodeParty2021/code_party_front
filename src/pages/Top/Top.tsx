import React from "react";
import { Link } from "react-router-dom";

type Prop = {};

export const Top: React.FC<Prop> = () => {
  return (
    <div>
      <h1>What is CodeParty</h1>
      <Link to="/start">Lets Play</Link>
    </div>
  );
};
