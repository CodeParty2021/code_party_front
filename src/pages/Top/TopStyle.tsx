import styled from "styled-components";
import { FlexGap } from "styles/FlexGap/FlexGap";

export const TopStyle = styled.div`
  width: 100vw;
  height: 100vh;

  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;
export const LogoStyle = styled.img`
  width: 665px;
  height: 306px;
`;

export const ButtonBox = styled.div`
  display: flex;
  flex-direction: column;
  ${FlexGap({ gap: "8px", direction: "column" })}
  margin-top:99px;
`;

export const CenterBox = styled.div`
  display: flex;

  align-items: center;
  flex-direction: column;
`;