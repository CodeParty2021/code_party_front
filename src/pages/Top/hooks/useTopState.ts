import { getAuth, signInAnonymously } from "firebase/auth";
import { useState } from "react";
import { useNavigate } from "react-router-dom";
export type IResponse = {
  FirstLoginBtnDisabled: boolean;
  FirstLoginBtnHandler: () => void;
};

export type IResponse2 = {
  NormalLoginBtnDisabled: boolean;
  NormalLoginBtnHandler: () => void;
};

export const useFisrtLoginTopState = (): IResponse => {
  const [FirstLoginBtnDisabled, setFirstLoginBtnDisabled] = useState(false);
  const navigate = useNavigate();

  const _FirtstLoginBtnHandler = () => {
    setFirstLoginBtnDisabled(true);
    signInAnonymously(getAuth()) //認証部分
      .then(() => {
        navigate("/event/set-name");
      })
      .catch(() => {
        setFirstLoginBtnDisabled(false);
      });
  };

  return {
    FirstLoginBtnDisabled: FirstLoginBtnDisabled,
    FirstLoginBtnHandler: _FirtstLoginBtnHandler,
  };
};

export const useNormalLoginTopState = (): IResponse2 => {
  const [NormalLoginBtnDisabled, setNormalLoginBtnDisabled] = useState(false);
  const navigate = useNavigate();

  const _NormalLoginBtnHandler = () => {
    setNormalLoginBtnDisabled(true);
    navigate("/start");
  };

  return {
    NormalLoginBtnDisabled: NormalLoginBtnDisabled,
    NormalLoginBtnHandler: _NormalLoginBtnHandler,
  };
};
