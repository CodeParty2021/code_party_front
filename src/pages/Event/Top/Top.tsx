import { Button } from "components/Button/Button";
import React from "react";
import { Background } from "../components/Background";
import { useTopState } from "./hooks/useTopState";
import { ButtonBox, CenterBox, LogoStyle, TopStyle } from "./TopStyle";

type Props = {};

export const EventTop: React.FC<Props> = () => {
  const { anonymousLoginBtnDisabled, anonymousLoginBtnHandler } = useTopState();
  return (
    <>
      <TopStyle>
        <CenterBox>
          <LogoStyle src="./logo454.png" />
          <ButtonBox>
            <Button
              color="pink"
              icon={null}
              onClick={anonymousLoginBtnHandler}
              size="L"
              status={anonymousLoginBtnDisabled ? "disabled" : "default"}
              value="いざスタート！"
            />
          </ButtonBox>
        </CenterBox>
      </TopStyle>
      <Background />
    </>
  );
};
