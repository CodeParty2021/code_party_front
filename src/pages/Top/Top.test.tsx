import React from "react";
import { shallow } from "enzyme";
import { Top } from "./Top";
import { IResponse, useTopState } from "./hooks/useTopState";

jest.mock("./hooks/useTopState");

const useTopStateMock = useTopState as jest.Mock;

const initialState: IResponse = {
  FirstLoginBtnDisabled: false,
  FirstLoginBtnHandler: () => {},
  NormalLoginBtnDisabled: false,
  NormalLoginBtnHandler: () => {},
  CreateUserBtnDisabled: false,
  CreateUserBtnHandler: () => {},
};

describe("<Top />", () => {
  beforeEach(() => {
    useTopStateMock.mockReturnValue({ ...initialState });
  });
  afterEach(() => {
    jest.resetAllMocks();
  });

  it("auth snapshot test", () => {
    const wrapper = shallow(<Top />);

    expect(wrapper.getElements()).toMatchSnapshot();
  });
});
