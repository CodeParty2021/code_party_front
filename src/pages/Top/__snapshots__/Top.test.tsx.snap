// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Top /> auth snapshot test 1`] = `
Array [
  <React.Fragment>
    <styled.div>
      <styled.div>
        <styled.img
          src="./logo.svg"
        />
        <styled.div>
          <Button
            color="pink"
            icon={null}
            onClick={[Function]}
            size="L"
            status="default"
            value="初めて遊ぶ！"
          />
          <Button
            color="blue"
            icon={null}
            onClick={[Function]}
            size="M"
            status="default"
            value="ログインして遊ぶ！"
          />
          <Button
            color="black"
            icon={null}
            onClick={[Function]}
            size="S"
            status="default"
            value="新しくアカウントを作る"
          />
        </styled.div>
      </styled.div>
    </styled.div>
    <Styled(StarBackground) />
  </React.Fragment>,
]
`;
